crypto/keys/keys.go

```diff
@@ -0,0 +1,104 @@
+package keys
+
+import (
+	fmt "fmt"
+
+	"github.com/cosmos/cosmos-sdk/codec/types"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	crypto "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/crypto/types/multisig"
+	proto "github.com/gogo/protobuf/proto"
+)
+
+var (
+	_ crypto.PubKey  = &Secp256K1PubKey{}
+	_ crypto.PrivKey = &Secp256K1PrivKey{}
+)
```

Comment:
to make sure `*Secp256K1PubKey` and `*Secp256K1PrivKey` implement `crypto.PubKey` and `crypto.PrivKey` interfaces respectively.

Reply:
Then let's move it to tests.

Reply:
That shouldn't be part of the tests, it's just a static check and it's extensively used throughout the whole code base.

Reply:
It's a test. Instead of having it in a global scope (even anonymous) we could move it to a test and make it clear what we are checking:

````
func TestPubKeyImplementers(t *testing.T) {
    ...
}
```

Having it in a global space is 
1. not helpful
2. confusing

Reply:
I don't find it confusing but helpful while reading the code and while writing the code too so you don't forget to implement some interface methods. I'm not sure how you could test that because if it doesn't implement the given interface, it's not even going to compile. As described here https://golang.org/doc/faq#guarantee_satisfies_interface, we could also have some `ImplementsFooer()` method to make it more explicit but then that would impact the whole code base and is something that should be addressed globally if we wanna keep things consistent, not just in this PR.

Reply:
I'm usually doing this in tests to not impact the "normal" code - in this approach the tests won't compile. 

But if you think having it in a "normal" code is cleaner then OK - I would then suggest to add a short comment (eg `// asserting interface implementation`)

Reply:
I'm pretty sure we take this approach (having implement assertions in normal code) throughout the cosmos-sdk codebase. I also agree that it's cleaner to have it in normal code. If we want to revisit this generically, it should be done as a separate refactor, not in this PR.